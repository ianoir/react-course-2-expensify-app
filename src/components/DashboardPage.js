// Set up the imports for the React Components
import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const DashboardPage = () => (
    <div>
        <ExpenseListFilters />    
        <ExpenseList />
    </div>
);

export default DashboardPage;