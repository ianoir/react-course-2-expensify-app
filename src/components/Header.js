// Set up the imports for the React Components
import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName="isActive">Dashboard</NavLink>
        <NavLink to="/create" activeClassName="isActive">Create an new expense</NavLink>
        <button onClick={startLogout}>Log out</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);