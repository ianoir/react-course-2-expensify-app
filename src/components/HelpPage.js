// Set up the imports for the React Components
import React from 'react';

const HelpPage = () => (
    <div>This is from my helpcomponent</div>
);

export default HelpPage;