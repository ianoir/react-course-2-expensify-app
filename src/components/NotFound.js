// Set up the imports for the React Components
import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        Error 404! - <Link to ="/">Go to Home</Link>
    </div>
);

export default NotFoundPage;