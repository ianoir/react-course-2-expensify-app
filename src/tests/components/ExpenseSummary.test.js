import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary'; 

// should render a copy of ExpenseSummary with 1 expense totalled
test('should render a copy of ExpenseSummary with 1 expense totalled', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot(); 
});


// should render a copy of ExpenseSummary with multiple expenses totalled
test('should render a copy of ExpenseSummary with multiple expenses totalled', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={20} expensesTotal={23511251} />);
    expect(wrapper).toMatchSnapshot(); 
});