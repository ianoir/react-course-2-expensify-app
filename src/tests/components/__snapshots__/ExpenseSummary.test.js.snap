// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a copy of ExpenseSummary with 1 expense totalled 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing
      <span>
         
      </span>
       
      expenses
       totalling
      <span>
         
        £2.35
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render a copy of ExpenseSummary with multiple expenses totalled 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing
      <span>
         
      </span>
       
      expenses
       totalling
      <span>
         
        £235,112.51
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
